hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -153,6 +159,14 @@ def _default_employee(self):
         store=True,
     )
 
+    @api.multi
+    @api.depends('date_start')
+    def _compute_sheet_name(self):
+        for sheet in self:
+            sheet.sheet_name = (
+                _('Week') + ' ' + str(sheet.date_start.isocalendar()[1])
```

Comment:
With this proposed solution, the name of the timesheet sheet becomes stored in the DB. This is nice because:

- in the list view it allows the user to sort the timesheet sheets by periods :+1: 
- it allows to easily create groupby filters on periods and doing more advanced filtering :+1: 

However this proposed solution has a side effect: sorting or grouping the timesheet sheets by periods will put together weeks (with same name) of past years, this will be confusing.

Would be possible to add the year in the `sheet_name`? Something like this:

```python
            sheet.sheet_name = _('Week') + ' %s %s' % (
                str(sheet.date_start.isocalendar()[1]),
                str(sheet.date_start.isocalendar()[0]),
            )
```

Another issue (but not caused by this PR) is that the monthly timesheet will continue to have the same name of a weekly timesheet. Not very nice, but luckily I don't have customers working with monthly timesheet sheets.

Reply:
I do have one in v10 that when migrated, will find the issue...

Reply:
Another side effect is when a user changes the language in the preferences.

Here is how it looks like when mixing english and italian:

![image](https://user-images.githubusercontent.com/7657635/55227986-3518f980-5219-11e9-99fc-3ff32fa49ff9.png)

Reply:
@astirpe I have idea how to fix this monthly/different weeks issue and will do that today. Adding year looks like a great idea to me, will do that as well. As for locale change - maybe we may use `context_dependent`? @pedrobaeza

Reply:
https://github.com/odoo/odoo/blob/44515bc7beae3b4c46b5bc2959c320695b51439d/odoo/fields.py#L320 this

Reply:
If you store something, then you lose translatable possibility, so maybe it's better to not do it stored.

Reply:
Okies, I'll un-store it :) Among other changes